katdal/lazy_indexer.py

```diff
@@ -434,38 +478,52 @@ def __init__(self, dataset, keep=(), transforms=()):
 
     @property
     def dataset(self):
+        """Array after first-stage indexing and transformation."""
         with self._lock:
             if self._dataset is None:
-                try:
-                    dataset = self._orig_dataset[self.keep]
-                except NotImplementedError:
-                    # Dask does not like multiple boolean indices: go one dim at a time
-                    dataset = self._orig_dataset
-                    for dim, keep_per_dim in enumerate(self.keep):
-                        dataset = da.take(dataset, keep_per_dim, axis=dim)
+                dataset = _dask_getitem(self._orig_dataset, self.keep)
                 for transform in self.transforms:
                     dataset = transform(dataset)
                 self._dataset = dataset
                 self._orig_dataset = None
             return self._dataset
 
-    def dask_getitem(self, keep):
-        """Index the array and return a dask array.
+    def add_transform(self, transform):
```

Comment:
This function needs a unit test, covering both cases (call before or after the lazy computation is done).

Reply:
Done.